kbase-extension/static/kbase/js/widgets/function_output/samples/kbaseSampleSet.js

```diff
@@ -0,0 +1,270 @@
+/*
+Sebastian Le Bras, David Lyon - April 2020
+*/
+define([
+    'kbwidget',
+    'jquery',
+    'kbaseAuthenticatedWidget',
+    'kbaseTabs',
+    'narrativeConfig',
+    'widgets/common/loadingMessage',
+    'widgets/common/errorMessage',
+    'widgets/function_output/samples/SampleSet',
+
+    // For effect.
+    'css!widgets/function_output/samples/KBaseSampleSet.css',
+], (
+    KBWidget,
+    $,
+    kbaseAuthenticatedWidget,
+    kbaseTabs,
+    Config,
+    LoadingMessage,
+    ErrorMessage,
+    SampleSet
+) => {
+    'use strict';
+
+    const SERVICE_TIMEOUT = 60000;
+
+    function formatField(value, schema) {
+        if (!schema) {
+            return $(`<span>`).text(String(value)).attr('title', String(value));
+        }
+        switch (schema.type) {
+            case 'string':
+                switch (schema.format) {
+                    case 'url':
+                        return $(`<a href='${value}' target='_blank'>`)
+                            .text(value)
+                            .attr('title', value);
+                    case 'ontology-term':
+                        return $(
+                            `<a href='/#ontology/term/${schema.namespace}/${encodeURIComponent(
+                                value
+                            )}' target='_blank'>`
+                        )
+                            .text(value)
+                            .attr('title', value);
+                    default:
+                        return $(`<span>`).text(value).attr('title', value);
+                }
+            case 'number':
+                if ('formatting' in schema.kbase) {
+                    return $(`<span>`)
+                        .text(Intl.NumberFormat('en-US', schema.kbase.formatting).format(value))
+                        .attr('title', value);
+                }
+                return $(`<span>`).text(String(value)).attr('title', String(value));
+
+            default:
+                return $(`<span>`).text(String(value)).attr('title', String(value));
+        }
+    }
+
+    return KBWidget({
+        name: 'kbaseSampleSetView',
+        parent: kbaseAuthenticatedWidget,
+        version: '1.0.0',
+        options: {
+            pageLimit: 10,
+            default_blank_value: '',
+        },
+
+        init: function (options) {
+            this._super(options);
+
+            this.obj_ref = this.options.upas.id;
+
+            this.model = new SampleSet({
+                workspaceURL: Config.url('workspace'),
+                sampleServiceURL: Config.url('sample_service'),
+                token: this.authToken(),
+                timeout: SERVICE_TIMEOUT,
+                ref: this.options.upas.id,
+            });
+
+            // Render
+            this.render();
+
+            return this;
+        },
+
+        loggedInCallback: function () {
+            this.render();
```

Comment:
Having `this.render()` here and in `init` will probably do the rendering twice.

IMO, the utility of the `kbaseAuthenticatedWidget` (getting auth tokens) is pretty much already included in the Runtime module. 

Though, really, as long as `this.authToken()` works in the `init` function, you can just skip the `loggedinCallback` and save a render cycle.